.azure-pipelines-steps.yml

```diff
@@ -0,0 +1,50 @@
+# Build steps common to all platforms
+
+steps:
+- task: NodeTool@0
+  inputs:
+    versionSpec: $(node_version)
+  displayName: 'Install Node.js'
+
+- task: Npm@1
+  inputs:
+    command: custom
+    customCommand: i -g npm
+  displayName: 'Update npm'
+  condition: eq(variables['node_version'], '6.x')
+
+- task: Npm@1
```

Comment:
This is probably more personal preference, but I avoid the "npm" task because it doesn't add much value, requires more lines (i.e. increases complexity), and you usually then end up with a mix of script and NPM task steps invoking NPM in different ways anyway. Like you have done, I like to have separate tasks for each NPM command (improves readability of the status/logs), but will just use "script" for each, for example:

```yaml
- script: npm ci
  displayName: 'Install dependencies'
```

Reply:
I could avoid it here `npm ci`, but not the `npm i -g npm` that we need to get the `ci` command available. When I do it straightforwardly on a script step, it doesn't update the toolcache version. I tried different combinations, and had to maintain it finally through `Npm@1`